python/ray/serve/tests/unit/test_llm_imports.py

```
@@ -1,13 +1,25 @@
 import sys
 import pytest
+from packaging import version
+import pydantic
 
+PYDANTIC_VERSION = version.parse(pydantic.__version__)
 
+
+@pytest.mark.skipif(
+    PYDANTIC_VERSION < version.parse("2.0"),
+    reason="ray.serve.llm requires pydantic>=2.0",
```

Can be in a follow up, we should probably update requirements and/or log error when users are using pydantic 1

---

this test should be moved to llm.serve test because it needs all the llm dependencies now that we don't have the try-catch thingy.

---

Nono, this is to test using them in serve's namespace. And we are not testing the logics implemented, just the import behavior. If users import those modules from serve, should we expect them to fail with some error or should we expect them to fail silently that's what it's testing.

---

if expect them to fail the just use `with pytest.raises(ImportError):` or something similar

---

I C. ok I change how we view things. 
Then it should be an ImportError